### PR TITLE
Infrastructure "Packaged Ultica in a git submodule repository"

### PR DESCRIPTION
**Pull request by int-ua**
#### Summary

SUMMARY: Infrastructure "Packaged Ultica moved into a separate repository and linked back with git submodule"

#### Purpose of change

Decrease default local clone size, decrease history size, automate including latest Ultica version.

#### Describe the solution

Created a new GitHub organization for managing tilesets at https://github.com/CDDA-Tilesets 
It is currently co-owned by  @I-am-Erk,  me, @Fris0uman and @John-Candlebury. Other tileset authors who follow licensing restrictions and CDDA maintainers are welcome to join. I'm also ready to transfer ownership to CDDA maintainers completely to reduce the risk of "supply chain" attacks. Or the setup can be replicated within the CleverRaven org.

:question: Will the build scripts include the `gfx/UltimateCataclysm/.git` folder into the release?

Note that `git submodule` by default is in detached HEAD state because a specific commit is referenced. Try `git reset --hard origin/main` from withing submodule to go to the latest commit. This will change the referenced commit and you will have the ability to commit new reference from outside the submodule directory.

#### Describe alternatives you've considered

Not changing default branch from `master` to `main`. But I like the idea and did it.

Added later: I believe it's also possible to entirely implement getting latest version directly from Ultica repo and feeding it to `compose.py` with GitHub actions or the build script but that would create additional load on CI and it's already taking hours to run on almost any PR. And it _may_ complicate things more (compared to having a git submodule) for people who want to have a copy of Ultica in their local clone.

#### Testing

AFAIU needs to be tested by the GitHub infrastructure directly.

#### Additional context

#28707

https://github.com/I-am-Erk/CDDA-Tilesets/issues/343

https://discord.com/channels/552510581161066497/552510581782085654/789461765230297098

### CODE REVIEW

**Comment by int-ua** [comment]
```
E: The repository 'https://apt.postgresql.org/pub/repos/apt bionic-pgdg Release' no longer has a Release file.
```
(sigh)

**Comment by int-ua** [comment]
I think I'll also try to preserve the commit history of that folder in the new repo. Mostly to show that it's possible.

**Comment by I-am-Erk** [comment]
This is a big deal and what I consider one of the ultimate goals of moving all our tilesets to a separate repo. If it works as advertised it allows way more flexible updating of tilesets while also reducing unnecessary bloat by storing a bunch of graphics in the main repo.

**Comment by int-ua** [comment]
Listed another possible alternative in the OP comment.

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -104,6 +104,16 @@ Helpful pages:
         $ git remote add -f upstream https://github.com/CleverRaven/Cataclysm-DDA.git
         # Assigns the original repository to a remote called "upstream"
 
+5. (optional) Download the UltimateCataclysm tileset submodule
+
+        $ git submodule update --init --remote --no-fetch gfx/UltimateCataclysm/
```
Comment (int-ua) [inline .github/CONTRIBUTING.md:109]: I need to check again if `--no-fetch` is beneficial here.

**Comment by actual-nh** [comment]
Status?

**Comment by int-ua** [comment]
Note to self: check if it will set the `-dirty` flag.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -104,6 +104,16 @@ Helpful pages:
         $ git remote add -f upstream https://github.com/CleverRaven/Cataclysm-DDA.git
         # Assigns the original repository to a remote called "upstream"
 
+5. (optional) Download the UltimateCataclysm tileset submodule
+
+        $ git submodule update --init --remote gfx/UltimateCataclysm
+        # Will get you the latest available commit in the default branch (main).
+        # If this commit has not been recorded by the CDDA repository yet
+        # it will have to be committed or ignored
```
Comment (int-ua) [inline .github/CONTRIBUTING.md:112]: Is there a better way? Didn't check yet.

**Comment by int-ua** [comment]
This PR can go, I think. I'll leave the branch in case anyone wants to retry :)

**Pull request closed without merging**